var/spack/repos/builtin/packages/parallel-netcdf/package.py

```diff
@@ -95,13 +95,27 @@ def configure_args(self):
         args += self.enable_or_disable('cxx')
         args += self.enable_or_disable('fortran')
 
+        flags = {
+            'CFLAGS': [],
+            'CXXFLAGS': [],
+            'FFLAGS': [],
+            'FCFLAGS': [],
+        }
+
         if '+pic' in self.spec:
-            args.extend([
-                'CFLAGS='   + self.compiler.cc_pic_flag,
-                'CXXFLAGS=' + self.compiler.cxx_pic_flag,
-                'FFLAGS='   + self.compiler.f77_pic_flag,
-                'FCFLAGS='  + self.compiler.fc_pic_flag,
-            ])
+            flags['CFLAGS'].append(self.compiler.cc_pic_flag)
+            flags['CXXFLAGS'].append(self.compiler.cxx_pic_flag)
+            flags['FFLAGS'].append(self.compiler.f77_pic_flag)
+            flags['FCFLAGS'].append(self.compiler.fc_pic_flag)
+
+        # https://github.com/Parallel-NetCDF/PnetCDF/issues/61
+        if self.spec.satisfies('%gcc@10:'):
+            flags['FFLAGS'].append('-fallow-argument-mismatch')
+            flags['FCFLAGS'].append('-fallow-argument-mismatch')
+
+        for key, value in flags.items():
```

Comment:
Maybe:
```suggestion
        for key, value in sorted(flags.items()):
```
to ensure consistent iterations across different runs?

Reply:
Done!